src/core/namespace.c

```diff
@@ -1346,14 +1346,68 @@ static int mount_mqueuefs(const MountEntry *m) {
         return 1;
 }
 
+static VeritySettings* verities_find_by_roothash(
+                VeritySettings *haystack,
+                size_t n_haystack,
+                const VeritySettings *needle) {
+
+        assert(haystack || n_haystack == 0);
+        assert(needle);
+
```

Comment:
one more tweak. can you return NULL here immediately if needle->root_hash_size is zero? the field is optional after all, and this shouldn't mean we consider it identical to another layer that has no roothash

Reply:
done